src/FSharpVSPowerTools.Core/Utils.fs

```diff
@@ -3,6 +3,24 @@
 open System
 open System.Diagnostics
 
+
+[<AutoOpen>]
+module Prelude =
+    /// obj.ReferenceEquals
+    let inline (==) a b = obj.ReferenceEquals(a, b)
+    /// LanguagePrimitives.PhysicalEquality
+    let inline (===) a b = LanguagePrimitives.PhysicalEquality a b
+    let inline debug msg = Printf.kprintf Debug.WriteLine msg
+    let inline fail msg = Printf.kprintf Debug.Fail msg
+    let inline isNull v = match v with | null -> true | _ -> false
+    let inline isNotNull v = v |> (not << isNull)
+    let inline dispose (disposable:#IDisposable) = disposable.Dispose ()
+    let inline (+=) (a1:'a byref) (a2:'a) = a1 <- a1 + a2
+    let inline (-=) (a1:'a byref) (a2:'a) = a1 <- a1 - a2
+
+    type 'a List with
+        member x.Last = x.[x.Length-1]
```

Comment:
Here are results on my machines:

> last 0.004ms
recursive last 0.002ms
  Ratio:  2.0
val it : unit = ()
> last 0.037ms
recursive last 0.021ms
  Ratio:  1.761904762
val it : unit = ()
> last 0.383ms
recursive last 0.222ms
  Ratio:  1.725225225
val it : unit = ()
> last 4.576ms
recursive last 2.495ms
  Ratio:  1.834068136
val it : unit = ()
>

It sounds reasonable wrt the dominant cost of list traversal (2:1 between the two implementations).